### ISSUE TITLE
iOS - OTSession.streamPropertyChanged not being called

### ISSUE THREAD

**Issue by evillemez** [opentok/opentok-react-native#269]
# Bug Report

**Current behavior**

With version `0.9.8` on iOS the `OTSession.streamPropertyChanged` event handler is not called for locally published streams.

**Steps to reproduce**

* start call on ios
* change value for either `OTPublisher.publishAudio` or `OTPublisher.publishVideo`
* note that a registered event listener for `OTSession.streamPropertyChanged` is not called (on Android it will be called)

**What is the current *bug* behavior?**

On iOS the `OTSession.streamPropertyChanged` handler is not called for locally published streams.  On Android it is called both for subscribed streams from remote users, as well as streams published by the local user.

**What is the expected *correct* behavior?**

The `OTSession.streamPropertyChanged` event handler should be called when the locally published stream has changed.

**Comment by msach22** [comment]
@evillemez Thanks for filing the issue here. In the iOS SDK, there is no `streamPropertyChanged` event for any stream. To emulate the JS event, we attach KVOs to the incoming streams and when those values change, we dispatch the `streamPropertyChanged` event. The Android SDK works a little differently because it has events for when the stream property changes. We can either attach KVOs to the publisher stream on iOS or maintain a local state because the app should know if it's publishing audio or video because of the properties you set when initializing the publisher or when you change these later down the line. LMK what you think.

**Comment by evillemez** [comment]
@msach22 Hmm, ok, interesting.  I guess my question is, for Android, is it theoretically possible for `Publisher.publishAudio` to be true, and the published streams `stream.hasAudio` to false?  In theory I would assume it shouldn't be possible, but I have no understanding of what's going on internally in the native SDK, so maybe there could be a case where you're trying to publish the audio or video but the published stream still doesn't have it for some reason, perhaps a malfunctioning camera/mic.

IF that case is theoretically possible on either platform, then I would go with the KVO approach in iOS.  If both platforms are actually just internally looking at the the values of the publisher to determine if the published stream has audio or video, then I would go with the latter approach if it's simpler.  If the library tracked the publisher values internally and fired the `streamPropertyChanged` event on iOS, then the behavior would be consistent from the caller's standpoint.

That said, my understanding of the native implementations is almost non-existent, so I'm not confident my instinct here is good.

**Comment by msach22** [comment]
@evillemez Apologies for the delay on this, I just rant the streamPropertyChanged event for iOS and I'm getting the event without any issues. 

Here's the sample code that I have for this:

```
import React, { Component } from 'react';
import { View } from 'react-native';
import { OTSession, OTPublisher, OTSubscriber } from 'opentok-react-native';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.apiKey = '';
    this.sessionId = '';
    this.token = '';
    this.sessionHandlers = {
      streamPropertyChanged: event => {
        console.log('event', event);
      }
    }
  }
  render() {
    return (
      <View style={{ flex: 1, flexDirection: 'row' }}>
        <OTSession apiKey={this.apiKey} sessionId={this.sessionId} token={this.token} eventHandlers={this.sessionHandlers}>
          <OTPublisher style={{ width: 100, height: 100 }} />
          <OTSubscriber style={{ width: 100, height: 100 }} />
        </OTSession>
      </View>
    );
  }
}
``` 

Please fill in the credentials and add a second participant using the [OpenTok Playground Tool](https://tokbox.com/developer/tools/playground/)

**Comment by evillemez** [comment]
Well... that's confusing. I'll retry testing this and see if I uncover anything different.  You were testing on `0.9.8` or `0.10.1`?

**Comment by msach22** [comment]
Yea, not sure why that's happening for you. I checked using `v0.10.2`.

**Comment by evillemez** [comment]
Do you think the update of the native SDKs changed the behavior then?

**Comment by msach22** [comment]
@evillemez My sincere apologies. In my test, I was only toggling the stream properties for the remote publisher. Please ignore my comment before.

I'm submitting a PR to add this feature.

**Comment by msach22** [comment]
This has been released with v0.10.3
